src/gui/CemuApp.cpp

```diff
@@ -231,33 +209,44 @@ int CemuApp::FilterEvent(wxEvent& event)
 	return wxApp::FilterEvent(event);
 }
 
-std::vector<const wxLanguageInfo*> CemuApp::GetAvailableLanguages()
-{
-	const auto path = ActiveSettings::GetDataPath("resources");
-	if (!exists(path))
-		return {};
-	
-	std::vector<const wxLanguageInfo*> result;
-	for (const auto& p : fs::directory_iterator(path))
-	{
-		if (!fs::is_directory(p))
-			continue;
+std::vector<const wxLanguageInfo *> CemuApp::GetLanguages() const {
+	std::vector availableLanguages(m_availableTranslations);
+	availableLanguages.insert(availableLanguages.begin(), wxLocale::GetLanguageInfo(wxLANGUAGE_ENGLISH));
+	return availableLanguages;
+}
 
-		const auto& path = p.path();
-		auto filename = path.filename();
+void CemuApp::LocalizeUI()
+{
+	std::unique_ptr<wxTranslations> translationsMgr(new wxTranslations());
+	m_availableTranslations = GetAvailableTranslationLanguages(translationsMgr.get());
 
-		const auto* lang_info = wxLocale::FindLanguageInfo(filename.c_str());
-		if (!lang_info)
-			continue;
+	const sint32 configuredLanguage = GetConfig().language;
+	bool isTranslationAvailable = std::any_of(m_availableTranslations.begin(), m_availableTranslations.end(),
+											  [configuredLanguage](const wxLanguageInfo* info) { return info->Language == configuredLanguage; });
+	if (configuredLanguage == wxLANGUAGE_DEFAULT || isTranslationAvailable)
+	{
+		translationsMgr->SetLanguage(static_cast<wxLanguage>(configuredLanguage));
+		translationsMgr->AddCatalog("cemu");
 
-		const auto language_file = path / "cemu.mo";
-		if (!fs::exists(language_file))
-			continue;
+		if (translationsMgr->IsLoaded("cemu") && wxLocale::IsAvailable(configuredLanguage))
+			m_locale.Init(configuredLanguage);
 
-		result.emplace_back(lang_info);
+		// This must be run after wxLocale::Init, as the latter sets up its own wxTranslations instance which we want to override
+		wxTranslations::Set(translationsMgr.release());
 	}
+}
 
-	return result;
+std::vector<const wxLanguageInfo*> CemuApp::GetAvailableTranslationLanguages(wxTranslations* translationsMgr)
+{
+	wxFileTranslationsLoader::AddCatalogLookupPathPrefix(ActiveSettings::GetDataPath("resources").generic_string());
```

Comment:
I don't trust `.generic_string()` to do the right conversion here in all situations. We have a helper which avoids implementation defined behavior: `wxString wxHelper::FromPath(const fs::path& path)`.

Reply:
Ok! Now it should be fine